scripts/demo/generate-demo.sh

```diff
@@ -0,0 +1,217 @@
+    
+    # Check for help flag
+    if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+        usage
+    fi
+    
+    # Validate demo script exists
+    if [[ ! -f "$DEMO_SCRIPT" ]]; then
+        error "Demo script not found: $DEMO_SCRIPT"
+    fi
+    
+    local missing_tools=()
+    
+    if ! command -v asciinema &> /dev/null; then
+        missing_tools+=("asciinema")
+    fi
+    
+    if ! command -v svg-term &> /dev/null; then
+        missing_tools+=("svg-term")
+    fi
+    
+    if ! command -v kind &> /dev/null; then
+        missing_tools+=("kind")
+    fi
+    
+    if ! command -v kubectl &> /dev/null; then
+        missing_tools+=("kubectl")
+    fi
+    
+    if [ ${#missing_tools[@]} -ne 0 ]; then
+        error "Missing required tools:\n$(printf '  - %s\n' "${missing_tools[@]}")"
+    fi
+    
+    log "All prerequisites are installed ✓"
+}
+
+setup_cluster() {
+    log "Setting up Kind cluster for demo..."
+    
+    # Use the e2e infrastructure to set up the cluster
+    export KIND_CLUSTER="kubebuilder-demo"
+    
+    # Build kubebuilder, fetch tools, and install kind if needed
+    build_kb
+    fetch_tools
+    install_kind
+    
+    # Create the cluster using the e2e setup function
+    create_cluster ${KIND_K8S_VERSION}
+    
+    log "Cluster connection verified ✓"
+}
+
+record_demo() {
+    local recording_dir="/tmp/kb-demo-recording"
+    
+    log "Cleaning up any previous recording files..."
+    rm -rf "$recording_dir"
+    mkdir -p "$recording_dir"
+    
+    log "Starting demo recording for '${DEMO_NAME}' in 3 seconds..."
+    sleep 3
+    
+    cd "$recording_dir"
+    asciinema rec \
+        --command "$DEMO_SCRIPT" \
+        --env "DEMO_AUTO_RUN=1" \
+        --title "Kubebuilder Demo: ${DEMO_NAME}" \
+        --idle-time-limit 2 \
+        "${DEMO_NAME}.cast"
+}
+
+convert_to_svg() {
+    local recording_dir="/tmp/kb-demo-recording"
+    local version="$1"
+    local svg_file="${PROJECT_ROOT}/docs/gif/${DEMO_NAME}.${version}.svg"
+    
+    log "Converting recording to SVG..."
+    svg-term \
+        --in="${recording_dir}/${DEMO_NAME}.cast" \
+        --out="$svg_file" \
+        --window \
+        --width=120 \
+        --height=30
+    
+    log "Demo updated! New file: docs/gif/${DEMO_NAME}.${version}.svg"
+    return 0
+}
+
+update_readme() {
+    local version="$1"
+    
+    # Only update README for the default kb-demo
+    if [[ "$DEMO_NAME" != "kb-demo" ]]; then
+        log "Skipping README update for custom demo '${DEMO_NAME}'"
+        return 0
+    fi
+    
+    log "Updating README.md with new demo..."
+    if [[ "$OSTYPE" == "darwin"* ]]; then
+        # macOS
+        sed -i '' "s|docs/gif/kb-demo\.v[^)]*\.svg|docs/gif/kb-demo.${version}.svg|g" "${PROJECT_ROOT}/README.md"
+    else
+        # Linux
+        sed -i "s|docs/gif/kb-demo\.v[^)]*\.svg|docs/gif/kb-demo.${version}.svg|g" "${PROJECT_ROOT}/README.md"
+    fi
+    
+    log "README.md updated with new demo file ✓"
```

Comment:
Is the script really generec and allow us to create demos for any place/doc? 
Or is it creating only for the README